src/views/states/list/utilts.ts

```diff
@@ -57,3 +57,12 @@ export const searchInterfaceByIP = (
 
   return addresses?.some((address) => address?.toLowerCase().includes(searchIPAddress));
 };
+
+export const searchInterfaceByMAC = (
+  searchMACAddress: string,
+  iface: NodeNetworkConfigurationInterface,
+) => {
+  if (!searchMACAddress) return true;
+
+  return iface?.['mac-address']?.includes(searchMACAddress) || false;
```

Comment:
Please use FILTER_TYPES.MAC_ADDRESS constant
and the `|| false` can be omitted

Reply:
i thought here to return false instead of `undefined` if there is no `mac-address`

the `FILTER_TYPES.MAC_ADDRESS` does not have anything to do with the `mac-address` property. It's the filter id :D

Reply:
you got the first if, which will return true when searchMACAddress is undefined or empty, btw I would use isEmpty there, since this is covered in the first if, the || false should be redundant, please check

Reply:
Thanks

Reply:
the || false is if the interface has no mac address. It's not covered by the first if. It's a different obj.